mantidproject/mantid
# Load correct grouping when field direction changes

Load correct grouping when changing run to one with a different field direction

Fixes a bug found in MuonAnalysis where, if a new run was loaded for the same instrument as previously but a different main field direction, the new grouping was not loaded.
Now checks for main field direction as well.

Refactored out the check for reloading the grouping and added tests.

**To test:**
_Data available in unit/doc test folders_
- Open Interfaces/Muon/Muon Analysis
- Load `MUSR00015189.nxs` (longitudinal field)
- Go to "Grouping Options" tab and edit the table to add another row - for example "other" with detectors 1-3
- Back on the "Home" tab, load `MUSR00015190.nxs` (longitudinal field)
  - Grouping should not be reloaded. The "Grouping Options" tab should be the same, including the new group you just added.
- Now load `MUSR00022725.nxs` (transverse field)
  - Grouping should be reloaded for this new configuration

Fixes #16224.

[Release notes](https://github.com/mantidproject/mantid/blob/065eca5e38c8087d2be23db654e8702729d5f8ab/docs/source/release/v3.7.0/muon.rst#id4) have been updated.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
